extensions/data-transfer/portability-data-transfer-google/src/main/java/org/dataportabilityproject/datatransfer/google/photos/GooglePhotosExporter.java

```diff
@@ -102,44 +95,42 @@ public GooglePhotosExporter(GoogleCredentialFactory credentialFactory) {
 
   private ExportResult<PhotosContainerResource> exportAlbums(
       TokensAndUrlAuthData authData, Optional<PaginationData> paginationData) {
-    int startItem = 1;
+    Optional<String> paginationToken = Optional.empty();
     if (paginationData.isPresent()) {
       String token = ((StringPaginationToken) paginationData.get()).getToken();
       Preconditions.checkArgument(
           token.startsWith(ALBUM_TOKEN_PREFIX), "Invalid pagination token " + token);
-      startItem = Integer.parseInt(token.substring(ALBUM_TOKEN_PREFIX.length()));
+      paginationToken = Optional.of(token.substring(ALBUM_TOKEN_PREFIX.length()));
     }
 
-    URL albumUrl;
-    UserFeed albumFeed;
+    AlbumListResponse albumListResponse;
 
     try {
-      albumUrl = new URL(String.format(URL_ALBUM_FEED_FORMAT, startItem, maxResults));
-      albumFeed = getOrCreatePhotosService(authData).getFeed(albumUrl, UserFeed.class);
-    } catch (ServiceException | IOException e) {
+      albumListResponse = getOrCreatePhotosInterface(authData).listAlbums(paginationToken);
+    } catch (IOException e) {
       return new ExportResult<>(ResultType.ERROR, e.getMessage());
     }
 
     PaginationData nextPageData = null;
-    List<GphotoEntry> entries = albumFeed.getEntries();
-    if (entries.size() == maxResults) {
-      int nextPageStart = startItem + maxResults;
-      nextPageData = new StringPaginationToken(ALBUM_TOKEN_PREFIX + nextPageStart);
+    GoogleAlbum[] googleAlbums = albumListResponse.getAlbums();
+    if (!Strings.isNullOrEmpty(albumListResponse.getNextPageToken())) {
+      nextPageData = new StringPaginationToken(ALBUM_TOKEN_PREFIX + albumListResponse.getNextPageToken());
     }
 
     ContinuationData continuationData = new ContinuationData(nextPageData);
-    List<PhotoAlbum> albums = new ArrayList<>(entries.size());
+    List<PhotoAlbum> albums = new ArrayList<>(googleAlbums.length);
```

Comment:
Why create this extra variable, and googleAlbums?  Can't we just iterate over albumListResponse.getAlbums()?

Reply:
We have to add the albums and photos to container resources as well.

Reply:
Oh sorry I didn't expand the diff, but you don't need googleAlbums right?